includes/data-port/import-tasks/class-sensei-import-associations.php

```diff
@@ -0,0 +1,413 @@
+
+			return false;
+		}
+
+		$result = wp_set_object_terms( $lesson_id, [ $term->term_id ], 'module' );
+		if ( is_wp_error( $result ) ) {
+			$add_warning_helper( $result->get_error_message(), $result->get_error_code() );
+
+			return false;
+		}
+
+		return $term->term_id;
+	}
+
+	/**
+	 * Helper method which gets a module by name and checks if the module can be applied to the lesson.
+	 *
+	 * @param string $module_name  The module name.
+	 * @param int    $course_id    Course ID.
+	 *
+	 * @return WP_Error|WP_Term  WP_Error when the module can't be applied to the lesson, WP_Term otherwise.
+	 */
+	private function get_lesson_module( $module_name, $course_id ) {
+		$module = get_term_by( 'name', $module_name, 'module' );
+
+		if ( ! $module ) {
+			return new WP_Error(
+				'sensei_data_port_module_not_found',
+				// translators: Placeholder is the term which errored.
+				sprintf( __( 'Module does not exist: %s.', 'sensei-lms' ), $module_name )
+			);
+		}
+
+		$course_modules = wp_list_pluck( wp_get_post_terms( $course_id, 'module' ), 'term_id' );
+
+		if ( ! in_array( $module->term_id, $course_modules, true ) ) {
+			return new WP_Error(
+				'sensei_data_port_module_not_part_of_course',
+				// translators: First placeholder is the term which errored, second is the course id.
+				sprintf( __( 'Module %1$s is not part of course %2$s.', 'sensei-lms' ), $module_name, $course_id )
+			);
+		}
+
+		return $module;
+	}
+
+	/**
+	 * Add a lesson to set a module for.
+	 *
+	 * @param int    $lesson_id   Lesson post ID.
+	 * @param string $module      Module name.
+	 * @param int    $line_number Line number for the lesson.
+	 * @param string $post_title  Import line post title.
+	 */
+	public function add_lesson_module( $lesson_id, $module, $line_number, $post_title ) {
+		$this->lesson_modules[ $lesson_id ] = [
+			$module,
+			$line_number,
+			$post_title,
+		];
+	}
+
+	/**
+	 * Add a course to assign lessons for.
+	 *
+	 * @param int    $course_id   Course post ID.
+	 * @param string $lessons     String list of references to lessons.
+	 * @param int    $line_number Line number for the lesson.
+	 * @param string $post_title  Import line post title.
+	 */
+	public function add_course_lessons( $course_id, $lessons, $line_number, $post_title ) {
+		$this->course_lessons[ $course_id ] = [
+			$lessons,
+			$line_number,
+			$post_title,
+		];
+	}
+
+	/**
+	 * Returns true if the task is completed.
+	 *
+	 * @return boolean
+	 */
+	public function is_completed() {
+		return isset( $this->total_tasks ) && 0 === $this->get_remaining_tasks();
+	}
+
+	/**
+	 * Returns the completion ratio of this task. The ration has the following format:
+	 *
+	 * {
+	 *
+	 *     @type integer $completed  Number of completed actions.
+	 *     @type integer $total      Number of total actions.
+	 * }
+	 *
+	 * @return array
+	 */
+	public function get_completion_ratio() {
+		$artificial_task_size = 10;
```

Comment:
This is so the progress doesn't shrink as more sub-tasks are added. This normalizes this task to be "10" steps. We might want to set this to a number based on the other task sizes. Right now, small imports may go from ~40% to finished.